lib/fs/fakefs.go

```diff
@@ -444,13 +504,25 @@ func (fs *fakefs) Rename(oldname, newname string) error {
 		return os.ErrNotExist
 	}
 
-	dst, ok := p1.children[filepath.Base(newname)]
-	if ok && dst.entryType == fakeEntryTypeDir {
-		return errors.New("is a directory")
+	dst, ok := p1.children[newKey]
+	if ok {
+		// in-place case-only directory renames will fail on insensitive systems
+		// they do fail on OS X, so that's expected
+		if dst.entryType == fakeEntryTypeDir {
+			return errors.New("is a directory")
+		}
```

Comment:
Shouldn't this be gated by fs.insens?
Presumably this should succeed on linux with case sensitivity?

Reply:
Nope. On sensitive filesystem this check is always requires, because there is no such thing as an in-place rename. `foo -> Foo` is always a rename to a new file, not the same one, and if `Foo` is an existing directory, we want to abort this, hence the check.